### ISSUE TITLE
[Question]: Automerging index with MongoDB

### ISSUE THREAD

**Issue by akabeera** [run-llama/llama_index#13472]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

I learned about auto merging index from the DLAI course and wanted to try implementing in my app.  My app use MongoDB VectorDB, and I retro-fitted the code from the source to find MongoDB's vector store?

```
    chunk_sizes = chunk_sizes or [1024, 512, 128]
    node_parser = HierarchicalNodeParser.from_defaults(chunk_sizes=chunk_sizes)
    nodes = node_parser.get_nodes_from_documents(documents)
    leaf_nodes = get_leaf_nodes(nodes)
    .
    .
    .
    atlas_vector_search = MongoDBAtlasVectorSearch(
        mongodb_client,
        db_name = DB_NAME,
        collection_name = DEST_COLLECTION_NAME,
        index_name = EMBEDDING_VECTOR_INDEX
    )

    vector_store_context = StorageContext.from_defaults(vector_store=atlas_vector_search)
    vector_store_context.docstore.add_documents(nodes)
    vector_store_index = VectorStoreIndex.from_documents(leaf_nodes, 
                                                         storage_context=vector_store_context, 
                                                         service_context=merging_context, 
                                                         show_progress=True)
```

But `VectorStoreIndex.from_documents` throws exception: 
```
'TextNode' object has no attribute 'get_doc_id'
```

I suspect I'm passing `nodes` instead of `documents` and nodes class don't have `get_doc_id`

Is there an alternative approach of implementing automerging index with mongo db?

**Comment by logan-markewich** [comment]
Don't use `from_documents()` with node objects

```
vector_store_index = VectorStoreIndex(
     nodes=leaf_nodes, 
     storage_context=vector_store_context, 
     service_context=merging_context, 
     show_progress=True
)
```

**Comment by akabeera** [comment]
Thanks, your suggestion worked.   I was able to run successfully by doing the following:

```
    vector_store_index = VectorStoreIndex(nodes=leaf_nodes,
                                          storage_context=vector_store_context,  
                                          service_context=merging_context, 
                                          show_progress=True)
```

Now I have a new question.  Looking at the embeddings in MongoDB, I don't see any metadata related to child or parent nodes.  It's just a flat list of embeddings.  Did I miss something else?

Unlike `SentenceWindowNodeParser` where I can specify metadatakeys like `window_metadata_key` and `original_text_metadata_key` I did not notice similar parameters for the `HierarchicalNodeParser`

Is the embeddings expected to be a flat list?  How can I the index was parsed successfully?

**Comment by akabeera** [comment]
@dosu
Thanks, your suggestion worked.   I was able to run successfully by doing the following:

```
    vector_store_index = VectorStoreIndex(nodes=leaf_nodes,
                                          storage_context=vector_store_context,  
                                          service_context=merging_context, 
                                          show_progress=True)
```

Now I have a new question.  Looking at the embeddings in MongoDB, I don't see any metadata related to child or parent nodes.  It's just a flat list of embeddings.  Did I miss something else?

Unlike `SentenceWindowNodeParser` where I can specify metadatakeys like `window_metadata_key` and `original_text_metadata_key` I did not notice similar parameters for the `HierarchicalNodeParser`

Is the embeddings expected to be a flat list?  How can I the index was parsed successfully?
